### DISCUSSION TITLE
Troubleshooting timeouts in LSP client

### DISCUSSION THREAD

**Discussion by mdbergmann** [scalameta/metals#2747] [General]
Hi.

I'm getting timeouts very often after opening up my project.

Received as (client logging):

```
[internal] (id:28) Thu Apr 29 09:19:16 2021:
(:timed-out :textDocument/documentHighlight :id 28 :params
            (:textDocument
             (:uri "file:///myscalafile.scala")
             :position
             (:line 165 :character 30)))
```

For a request:

```
[client-request] (id:28) Thu Apr 29 10:58:47 2021:
(:jsonrpc "2.0" :id 28 :method "textDocument/documentHighlight" :params
          (:textDocument
           (:uri "file:///,yscalafile.scala")
           :position
           (:line 316 :character 7)))
```

Is there a way to troubleshoot why the request times out?

**Comment by tgodzik** [comment]
That method doesn't really wait for anything, so it should return quickly. This might be that the CPU is just being used by indexing/compiling, which I sam sometimes cause it. However, document highlight is resent whenever cursor moves on any text, so this shouldn't be a large issue.

**Reply by mdbergmann** [reply]
I see this with Eglot on Emacs, where one text editor pane (buffer) gets timeouts for hovering, completions, etc. while another editor pane works OK.
I believe that Eglot create a connection per buffer. Which would mean that one connection works, but the other not.
Weird.

**Reply by mdbergmann** [reply]
I've tried lsp-mode on Emacs and it seems that close to all of my issues with hangs and timeouts are gone.

So it seems the issue lies with Eglot.
I will try to troubleshoot in that direction.
